keadm/cmd/keadm/app/cmd/util/helminstaller.go

```diff
@@ -275,17 +290,42 @@ func (cu *KubeCloudHelmInstTool) runHelmManifest(r *Renderer, stdout io.Writer)
 }
 
 // RunHelmInstall starts cloudcore deployment with the given flags
-func (cu *KubeCloudHelmInstTool) runHelmInstall(r *Renderer) error {
+func (cu *KubeCloudHelmInstTool) runHelmInstall(r *Renderer) (*release.Release, error) {
 	cf := genericclioptions.NewConfigFlags(true)
 	cf.KubeConfig = &cu.KubeConfig
 	cf.Namespace = &cu.Namespace
 
 	cfg := &action.Configuration{}
-	logFunc := func(format string, v ...interface{}) {
-		fmt.Println(fmt.Sprintf(format, v...))
-	}
+	// let the os.Stdout not print the details
+	logFunc := func(format string, v ...interface{}) {}
 	if err := cfg.Init(cf, cu.Namespace, "", logFunc); err != nil {
-		return err
+		return nil, err
+	}
+
+	// a flag to confim the install/upgrade action
+	var performInstall bool
+	_, err := cfg.Releases.Last(r.componentName)
```

Comment:
Make a quick request to check the release exists or not to do install or upgrade.